source/PlayServicesResolver/src/PlayServicesResolver.cs

```diff
@@ -928,32 +926,69 @@ public class AndroidSdkRootChangedArgs : EventArgs {
         /// Defaults to 1 second.</param>
         private static void ScheduleAutoResolve(double delayInMilliseconds = 1000.0) {
             lock (typeof(PlayServicesResolver)) {
-                if (!autoResolving) {
-                    RunOnMainThread.Cancel(autoResolveJobId);
-                    autoResolveJobId = RunOnMainThread.Schedule(
-                        () => {
-                            lock (typeof(PlayServicesResolver)) {
-                                autoResolving = true;
-                            }
-                            RunOnMainThread.PollOnUpdateUntilComplete(() => {
-                                if (EditorApplication.isCompiling) return false;
-                                // Only run AutoResolve() if we have a valid autoResolveJobId.
-                                // If autoResolveJobId is 0, ScheduleResolve()
-                                // has already been run and we should not run AutoResolve()
-                                // again.
-                                if (autoResolveJobId != 0) {
-                                    AutoResolve(() => {
-                                        lock (typeof(PlayServicesResolver)) {
-                                            autoResolving = false;
-                                            autoResolveJobId = 0;
+                if (autoResolving)
+					return;
```

Comment:
style: Could you move this to the previous line? 

We try to avoid:

if (foo)
  return;

in preference for

if (foo) {
  return;
}

or

if (foo) return;

The main issue with the continuation on a newline without braces is that other folks can come along and easily break the code with...

if (foo)
  do_something();
  return;